### PR TITLE
ID-623 Signed URL Query Params and Requester Pays Support

### PR DESCRIPTION
**Pull request by tlangs**
Ticket: https://broadworkbench.atlassian.net/browse/ID-623

What:

Make Sam use the workspace project as the requester-pays project. Also, inject the user email into the signed url for auditability.

Why:

Feature request from @nmalfroy. This lets TDR not need to sign its own URLs and leverages Sam to do it, reducing the need to pass around Service Account Keys (good for security!).

How:

Just pass a few more params to the workbench-libs implementation.

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've filled out the [Security Risk Assessment](https://sdarq.dsp-appsec.broadinstitute.org/jira-ticket-risk-assesment) (requires Broad Internal network access) and attached the result to the JIRA ticket

### CODE REVIEW

**Comment by tlangs** [comment]
Tests will fail until https://github.com/broadinstitute/workbench-libs/pull/1350 merges and the version of workbench-libs is updated.

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala [hunk 1]
```diff
@@ -641,12 +645,25 @@ class GoogleExtensions(
...
+      duration: Option[Long],
+      samRequestContext: SamRequestContext
+  ): IO[URL] =
     for {
       petServiceAccount <- createUserPetServiceAccount(samUser, project, samRequestContext)
       petKey <- googleKeyCache.getKey(petServiceAccount)
       serviceAccountCredentials = ServiceAccountCredentials.fromStream(new ByteArrayInputStream(petKey.getBytes()))
-      url <- googleStorageService.getSignedBlobUrl(bucket, name, serviceAccountCredentials).compile.lastOrError
+      time = duration.getOrElse(1L)
+      timeUnit = duration.map(_ => TimeUnit.MINUTES).getOrElse(TimeUnit.HOURS)
```
Comment (nmalfroy) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala]: This feels a little confusing no?  Why not just have
```suggestion
      time = duration.getOrElse(60L)
      timeUnit = TimeUnit.MINUTES
```

and timeUnit could just be a constant

**Review by nmalfroy — APPROVED** [review]
looks good!  I do think it's a little weird to have different units of time for the default but your call on addressing.  The functionality seems to do the thing

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala (hunk 1 above)
Comment (tlangs) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala]: Good point. Inlined the timeUnit and renamed the variable to be clearer.

**Comment by tlangs** [comment]
jenkins retest

**Review by Shakespeared — APPROVED** [review]

**Review by Ghost-in-a-Jar — APPROVED** [review]

**Pull request merged**